pkg/helpers/identifier_string_parser_test.go

```diff
@@ -91,4 +94,251 @@ func Test_ParseIdentifierString(t *testing.T) {
 		require.NoError(t, err)
 		containsAll(t, parts, expected)
 	})
+
+	t.Run("empty identifier", func(t *testing.T) {
+		input := `""`
+		expected := []string{""}
+
+		parts, err := ParseIdentifierString(input)
+
+		require.NoError(t, err)
+		containsAll(t, parts, expected)
+	})
+
+	t.Run("handled correctly double quotes", func(t *testing.T) {
+		input := `""."."".".".""."".""."".""."".""."""""`
+		expected := []string{"", `.".`, `.".".".".".".".""`}
+
+		parts, err := ParseIdentifierString(input)
+
+		require.NoError(t, err)
+		containsAll(t, parts, expected)
+	})
+}
+
+func Test_ParseAccountObjectIdentifier(t *testing.T) {
+	testCases := []struct {
+		Input    string
+		Expected sdk.AccountObjectIdentifier
+		Error    string
+	}{
+		{Input: ``, Error: "incompatible identifier: "},
+		{Input: "a\nb", Error: "incompatible identifier: a\nb"},
+		{Input: `a"b`, Error: "unable to read identifier: a\"b, err = parse error on line 1, column 2: bare \" in non-quoted-field"},
```

Comment:
is snowflake escaping such `"` or do we have to do this on the SDK level?

Reply:
Not sure If I understand correctly, but `a"b` is not allowed (in Snowflake and CSV parser). It would be legal in Snowflake and CSV parser if it would be enclosed by double quotes, and the quote inside should be escaped with another quote like so `"a""b"` (which results in `a"b` being saved in SF).

Reply:
yes, and in such a case, what will the Snowflake return: escaped or unescaped quote inside?

Reply:
I checked, and unfortunately, Snowflake will return this as unescaped, which will result in an error from the parser. Realistically speaking, I doubt anyone would like to put double quotes as part of their identifier, but I guess if Snowflake allows it, we should too 🙁. I checked further on grants how it will return multiple parts, and this time, it returned it as escaped (database and schema named as a"b` resulted in `"a" b." "a" b"` 👍). That makes me wonder if we should try and encode single-column identifiers with csv writer 🤔, but again, that's only to cover this case. Should I document it and leave it for now or do we want to have an implementation and example of this somewhere in code? e.g. database `<name>`

Reply:
As discussed, added a validation for now that would return an error whenever double quote is in the identifier.

Reply:
Did you do a follow-up issue for this?

Reply:
SNOW-1571674 added as in-code comment